superduper/backends/base/query.py

```diff
@@ -223,33 +223,35 @@ def updated_key(self, update):
         """Property setter."""
         self._updated_key = update
 
-    def dependencies(
+    def get_events(
         self,
+        ids: t.Sequence,
     ):
-        """List of dependencies."""
-        listeners = self.db.show('listener')
-        vector_indices = self.db.show('vector_index')
-        dependencies = []
-
-        def _check_query_match(listener, query):
-            return listener.depends_on(query)
-
-        for listener in listeners:
-            listener = self.db.listeners[listener]
-            if _check_query_match(listener, self):
-                dependencies.append(
-                    {'type_id': 'listener', 'identifier': listener.identifier}
-                )
+        """List of events.
 
-        for vi in vector_indices:
-            vi = self.db.vector_indices[vi]
-            listener = vi.indexing_listener
-            if _check_query_match(listener, self):
-                dependencies.append(
-                    {'type_id': 'vector_index', 'identifier': vi.identifier}
-                )
+        :param ids: List of ids.
+        """
+        listeners = [
+            self.db.listeners[identifier] for identifier in self.db.show('listener')
+        ]
+        vector_indices = [
+            self.db.vector_indices[identifier]
+            for identifier in self.db.show('vector_index')
+        ]
 
-        return dependencies
+        events = []
+
+        for component in listeners + vector_indices:
+            trigger_ids = component.trigger_ids(self, ids)
+            if trigger_ids:
+                events.append(
+                    {
+                        'type_id': component.type_id,
+                        'identifier': component.identifier,
+                        'ids': trigger_ids,
+                    }
+                )
+        return events
```

Comment:
Fix the compatibility bug between outputs query and queue.​⬤

Reply:
We should remove the need to load the whole component when doing this type of logic. These could be very large models.

Reply:
Yes, agreed, will change it after your PR merge

Reply:
This is a bigger task, let's do it later.

Reply:
OK